bot/exts/filtering/_ui/filter_list.py

```diff
@@ -0,0 +1,271 @@
+        self.embed = embed
+        self.confirm_callback = confirm_callback
+
+        self.settings_repr_dict = {name: to_serializable(value) for name, value in settings.items()}
+        populate_embed_from_dict(embed, self.settings_repr_dict)
+
+        self.type_per_setting_name = {setting: info[2] for setting, info in loaded_settings.items()}
+
+        edit_select = CustomCallbackSelect(
+            self._prompt_new_value,
+            placeholder="Select a setting to edit",
+            options=[SelectOption(label=name) for name in sorted(settings)],
+            row=0
+        )
+        self.add_item(edit_select)
+
+    @discord.ui.button(label="✅ Confirm", style=discord.ButtonStyle.green, row=1)
+    async def confirm(self, interaction: Interaction, button: discord.ui.Button) -> None:
+        """Confirm the content, description, and settings, and update the filters database."""
+        await interaction.response.edit_message(view=None)  # Make sure the interaction succeeds first.
+        try:
+            await self.confirm_callback(interaction.message, self.list_name, self.list_type, self.settings)
+        except ResponseCodeError as e:
+            await interaction.message.reply(embed=format_response_error(e))
+            await interaction.message.edit(view=self)
+        else:
+            self.stop()
+
+    @discord.ui.button(label="🚫 Cancel", style=discord.ButtonStyle.red, row=1)
+    async def cancel(self, interaction: Interaction, button: discord.ui.Button) -> None:
+        """Cancel the operation."""
+        await interaction.response.edit_message(content="🚫 Operation canceled.", embed=None, view=None)
+        self.stop()
+
+    def current_value(self, setting_name: str) -> Any:
+        """Get the current value stored for the setting or MISSING if none found."""
+        if setting_name in self.settings:
+            return self.settings[setting_name]
+        return MISSING
+
+    async def update_embed(
+        self,
+        interaction_or_msg: discord.Interaction | discord.Message,
+        *,
+        setting_name: str | None = None,
+        setting_value: str | None = None,
+    ) -> None:
+        """
+        Update the embed with the new information.
+
+        If `interaction_or_msg` is a Message, the invoking Interaction must be deferred before calling this function.
+        """
+        if not setting_name:  # Obligatory check to match the signature in the parent class.
+            return
+
+        self.settings[setting_name] = setting_value
+
+        self.embed.clear_fields()
+        new_view = self.copy()
+
+        try:
+            if isinstance(interaction_or_msg, discord.Interaction):
+                await interaction_or_msg.response.edit_message(embed=self.embed, view=new_view)
+            else:
+                await interaction_or_msg.edit(embed=self.embed, view=new_view)
+        except discord.errors.HTTPException:  # Various errors such as embed description being too long.
```

Comment:
Should we have some embed length trimming / limitation?

Reply:
Fair enough, I'll admit I was being lazy here.

Reply:
The issue was actually in a different file, the comment here was just a copy paste. b67e849